torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -778,6 +807,16 @@ c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupNCCL::WorkNCCL::
   return c10::make_intrusive<FutureNCCL>(shared_from_this(), outputs_);
 }
 
+void ProcessGroupNCCL::enqueue(
+    std::shared_ptr<ProcessGroupNCCL::WorkNCCL> work) {
+  {
+    std::lock_guard<std::mutex> lock(workVectorMutex_);
+    auto it = workVector_.end();
+    workVector_.emplace(it, std::move(work));
```

Comment:
curious, is this faster than `workVector_.emplace(std::move(work))`?

Reply:
I think `std::vector::emplace` requires a position argument, but we can use `std::vector::emplace_back` here instead.